packages/components/src/internal/components/samples/utils.tsx

```diff
@@ -278,16 +279,34 @@ export const getSampleTypeTemplateUrl = (
  * Provides sample wizard URL for this application.
  * @param targetSampleSet - Intended sample type of newly created samples.
  * @param parent - Intended parent of derived samples. Format SCHEMA:QUERY:ID
+ * @param selectionKey
+ * @param currentProductId
+ * @param targetProductId
  */
-export function getSampleWizardURL(targetSampleSet?: string, parent?: string): AppURL {
-    let url = NEW_SAMPLES_HREF;
+export function getSampleWizardURL(
+    targetSampleSet?: string,
+    parent?: string,
+    selectionKey?: string,
+    currentProductId?: string,
+    targetProductId?: string
+): string | AppURL {
+    let params = {},
+        url;
 
     if (targetSampleSet) {
-        url = url.addParam('target', targetSampleSet);
+        params['target'] = targetSampleSet;
     }
 
     if (parent) {
-        url = url.addParam('parent', parent);
+        params['parent'] = parent;
+    }
+
+    if (selectionKey) params['selectionKey'] = selectionKey;
+
+    if (currentProductId && targetProductId && currentProductId !== targetProductId) {
```

Comment:
I noticed this type of if statement in several places in this PR. It looks like if you would just always call `createProductUrlFromParts()` and pass through the props for the target and current productId, the `createProductUrl()` function does a very similar (or the same) check and would do the right thing. I didn't test it, but I think that would simplify these usages if you could remove this duplicate if/else check outside of `createProductUrlFromParts()`

Reply:
good point. updated.